python/src/eval.cc

```diff
@@ -0,0 +1,72 @@
+#include "internal/eval.hh"
+#include "internal/errors.hh"
+#include "internal/nix-to-python.hh"
+#include "internal/python-to-nix.hh"
+#include <Python.h>
+
+#include <cxxabi.h>
+#include <store-api.hh>
+
+namespace nix::python {
+
+const char * currentExceptionTypeName()
+{
+    int status;
+    auto res = abi::__cxa_demangle(abi::__cxa_current_exception_type()->name(), 0, 0, &status);
+    return res ? res : "(null)";
+}
+
+static PyObject * _eval(const char * expression, PyObject * vars)
+{
+    nix::Strings storePath;
+    nix::EvalState state(storePath, nix::openStore());
+
+    nix::Env * env = nullptr;
+    auto staticEnv = pythonToNixEnv(state, vars, &env);
+    if (!staticEnv) {
+        return nullptr;
+    }
+    auto staticEnvPointer = std::make_shared<nix::StaticEnv>(*staticEnv);
+    nix::Value v;
+
+
+    // FIXME: Doing this would breaks the test_GIL_case test case
+    // Release the GIL, so that other Python threads can be running in parallel
+    // while the potentially expensive Nix evaluation happens. This is safe
+    // because we don't operate on Python objects or call the Python/C API in
+    // this block
+    // See https://docs.python.org/3/c-api/init.html#thread-state-and-the-global-interpreter-lock
+    // Py_BEGIN_ALLOW_THREADS
```

Comment:
```suggestion
    Py_BEGIN_ALLOW_THREADS /* opens a block */
    Finally reacquireGIL([&] { Py_BLOCK_THREADS });
```

Reply:
What's this `Finally` here? It doesn't exist. Maybe I should just use a try-catch here instead?

Reply:
Finally is from `libutil/finally.hh`